File: day-01/part-2/jrmyy.py
from tool.runners.python import SubmissionPy


class JrmyySubmission(SubmissionPy):

    def run(self, s: str) -> int:
        # :param s: input in string format
        # :return: solution flag
        return sum(self.get_fuel(int(x)) for x in s.splitlines())

    def get_fuel(self, i: int, fuel_sum: int = 0) -> int:
        fuel_res = int(i) // 3 - 2
        if fuel_res < 0:
            return fuel_sum
        else:
            return self.get_fuel(fuel_res, fuel_sum + fuel_res)

File: day-01/part-2/jrmyy.ts
class Solution {

	// This is your puzzle input
	private readonly input: string;

	constructor(input: string) {
		this.input = input;
	}

	getFuel(value: number, fuelSum = 0): number {
		const fuelRes = Math.floor(value / 3) - 2;
		return fuelRes < 0 ? fuelSum : this.getFuel(fuelRes, fuelSum + fuelRes);
	}

	run(): number {
		return this.input.split('\n')
			.map(v => this.getFuel(parseInt(v, 10)))
			.reduce((acc, value) => acc + value, 0);
	}

	main() {
		const start = Date.now();
		const answer = this.run();

		console.log(`_duration:${(Date.now() - start).toString()}`);
		console.log(answer);
	}
}

// @ts-ignore
const solution = new Solution(process.argv[2]);
solution.main();
